presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparisonStrategies.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.tests.statistics;
+
+import com.google.common.collect.Range;
+
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class MetricComparisonStrategies
+{
+    private MetricComparisonStrategies() {}
+
+    public static MetricComparisonStrategy noError()
+    {
+        return absoluteError(0);
+    }
+
+    public static MetricComparisonStrategy absoluteError(double error)
+    {
+        return absoluteError(Range.closed(-error, error));
+    }
+
+    public static MetricComparisonStrategy absoluteError(Range<Double> errorRange)
+    {
+        return (actual, estimate) -> mapRange(errorRange, endpoint -> endpoint + actual)
+                .contains(estimate);
+    }
+
+    public static MetricComparisonStrategy defaultTolerance()
+    {
+        return relativeError(.1);
+    }
+
+    public static MetricComparisonStrategy relativeError(double error)
+    {
+        return relativeError(Range.closed(-error, error));
+    }
+
+    public static MetricComparisonStrategy relativeError(Range<Double> errorRange)
+    {
+        return (actual, estimate) -> mapRange(errorRange, endpoint -> (endpoint + 1) * actual)
+                .contains(estimate);
+    }
+
+    private static Range<Double> mapRange(Range<Double> range, Function<Double, Double> mappingFunction)
```

Comment:
I do not see a value. IMO the problem is with the fact that mapping function works on two domains: double and range. In my version I have a mapping function which maps range to a range with the usage of double to double function, in your version I would have a mapping function double to a range with usage of range to range function. 

Unless I missed something ; /